gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InstrumentatedGoogleHadoopFileSystem.java

```diff
@@ -321,10 +328,82 @@ public GhfsStorageStatistics getStorageStatistics() {
    */
   @Override
   public IOStatistics getIOStatistics() {
+    setHttpStatistics();
     return instrumentation != null ? instrumentation.getIOStatistics() : null;
   }
 
   public GhfsInstrumentation getInstrumentation() {
     return this.instrumentation;
   }
+
+  /** Set the Value for http get and head request related statistics keys */
+  private void setHttpStatistics() {
+    this.gcs = getGcsFs().getGcs();
+    try {
+      setRequestStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST);
+      setRequestStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_HEAD_REQUEST);
+      setRequestStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST);
+      setRequestStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_POST_REQUEST);
+      setRequestStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PATCH_REQUEST);
+      setRequestStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_DELETE_REQUEST);
+    } catch (Exception e) {
+      logger.atWarning().log("Error in getting statistics from gcsio", e);
+    }
+  }
+
+  /** Set the IOStatistics for http requests */
+  private void setRequestStatistics(GoogleCloudStorageStatistics key) {
+    GhfsStatistic statisticsKey = getKey(key);
+    clearStats(statisticsKey.getSymbol());
+    AtomicLong successKeyValue = gcs.getHttpStatistics(key);
+    AtomicLong failureKeyValue = gcs.getHttpStatistics(getFailureKey(key));
+    if (successKeyValue != null) {
+      incrementStatistic(statisticsKey, successKeyValue.longValue());
+      if (failureKeyValue != null) {
+        instrumentation.incrementFailureStatistics(statisticsKey, failureKeyValue.longValue());
```

Comment:
let's not differentiate success and failure stats. It should be driven by the key